server/src/main/kotlin/fi/oph/kitu/RestClientConfigurations.kt

```diff
@@ -0,0 +1,42 @@
+package fi.oph.kitu
+
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.core.StreamReadConstraints
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
+import org.springframework.web.client.RestClient
+
+@Configuration
+class RestClientConfigurations {
+    @Bean("restClientBuilderForLargeResponses")
+    fun restClientBuilderForLargeResponses(): RestClient.Builder =
+        RestClient
+            .builder()
```

Comment:
Huom: https://docs.spring.io/spring-boot/reference/actuator/tracing.html#actuator.micrometer-tracing.propagating-traces

```suggestion
    fun restClientBuilderForLargeResponses(restClientBuilder: RestClient.Builder): RestClient.Builder =
        restClientBuilder
```

Reply:
Tämä @Configuration kylläkin yliajaa tuon oletus-konfiguraation joka luo tuon RestClient.Builderin, joten en ole varma toimiiko tämä tai tuleeko tuosta ikikiersiö.

Reply:
Tuosta muutoksesta tulee nykyisellä build-virhe:
> Relying upon circular references is discouraged and they are prohibited by default. Update your application to remove the dependency cycle between beans. As a last resort, it may be possible to break the cycle automatically by setting spring.main.allow-circular-references to true.

Reply:
Äh, mitenköhän toi sitten pitäisi tehdä. Pitää tutkia asiaa. Voi mergetä näin mutta luultavasti sitten ei tule sub-spania luultavasti tuolle kutsulle opentelemetryyn.

Reply:
Hmm ei kai tuossa pitäis syntyä mitään riippuvuussykliä, jos `@Qualifier`/`@Bean`-annotaatiot on oikein paikallaan. Ihmettelen kohta tätäkin, että mikä homma :+1:

Reply:
Kiersin tän muuttamalla tämän custom-konfiguraation builderin extension-metodiksi. Näin voidaan käyttää automaagisesti konfiguroitua builderia, ja kutsua tälle metodia joka konfiguroi builderin isoille datamäärille. "Ylimääräinen" large-request bean putoaa pois, jolloin autoconfigured bean on oletuksena kaikkialla käytössä.

Mitä tätä ihmetellessä tuli kaiveltua, johtuen siitä miten tuon rest clientin automaattinen konfiguraatio toimii, vanha toteutus olisi mahdollisesti käyttänyt **kaikkialla** isojen pyyntöjen rest clienttia, `Qualifier`-annotaatioista huolimatta :sweat_smile: